test/backup.ts

```diff
@@ -0,0 +1,516 @@
+      const BACKUP_INFO_RESPONSE: GetMetadataResponse = [
+        {
+          expireTime: BACKUP_EXPIRE_TIME.toStruct(),
+        },
+      ];
+      backup.getMetadata = async () => BACKUP_INFO_RESPONSE;
+
+      const result = await backup.getExpireTime();
+      assert.deepStrictEqual(result, BACKUP_EXPIRE_TIME);
+    });
+
+    it('should throw errors', async () => {
+      const err = {code: grpc.status.INTERNAL};
+      backup.getMetadata = async () => {
+        throw err;
+      };
+
+      try {
+        await backup.getExpireTime();
+        assert.fail('Should have rethrown error');
+      } catch (thrown) {
+        assert.deepStrictEqual(thrown, err);
+      }
+    });
+  });
+
+  describe('exists', () => {
+    it('should return true when backup info indicates backup exists', async () => {
+      const BACKUP_INFO_RESPONSE: GetMetadataResponse = [{}];
+      backup.getMetadata = async () => BACKUP_INFO_RESPONSE;
+
+      const result = await backup.exists();
+      assert.strictEqual(result, true);
+    });
+
+    it('should return false when backup does not exist', async () => {
+      backup.getMetadata = async () => {
+        throw {code: grpc.status.NOT_FOUND};
+      };
+
+      const result = await backup.exists();
+      assert.strictEqual(result, false);
+    });
+
+    it('should rethrow other errors', async () => {
+      const err = {code: grpc.status.INTERNAL};
+      backup.getMetadata = async () => {
+        throw err;
+      };
+
+      try {
+        await backup.exists();
+        assert.fail('Should have rethrown error');
+      } catch (thrown) {
+        assert.deepStrictEqual(thrown, err);
+      }
+    });
+  });
+
+  describe('updateExpireTime', () => {
+    const NEW_EXPIRE_TIME = new PreciseDate(1977, 1, 10);
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        backup: {
+          name: BACKUP_FORMATTED_NAME,
+          expireTime: NEW_EXPIRE_TIME.toStruct(),
+        },
+        updateMask: {
+          paths: ['expire_time'],
+        },
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'updateBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME);
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME, options, assert.ifError);
+    });
+
+    it('should return backup object with updated expire time', () => {
```

Comment:
I believe we should simplify this test, only for the sake of clarity. As it appears now to a future contributor or maintainer, the method being tested "returns a backup object with updated expire time". In reality, it only "returns what the gax API returns". I would avoid forcing the mental complexity of figuring that out, even though both are correct.

We can just test that our simulated "error" and "response" from gax are exactly what we return.

Reply:
Thanks for the suggestion. It's much cleaner now.